src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java

```diff
@@ -181,7 +181,9 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .requestMatchers("/content/**", "/i18n/*.json", "/logo/*").permitAll()
                 .requestMatchers("/management/info", "/management/health").permitAll()
                 .requestMatchers("/api/admin/**").hasAuthority(Role.ADMIN.getAuthority())
+                .requestMatchers("/api/{version:v\\d+}/admin/**").hasAuthority(Role.ADMIN.getAuthority())
                 .requestMatchers("/api/public/**").permitAll()
+                .requestMatchers("/api/{version:v\\d+}/public/**").permitAll()
```

Comment:
Should we expect more of these duplicates with and without `version`? We could also use an option in the regex like this: `/api/({version:v\\d+}/)?public/**`, can't we?

Reply:
No, I don't expect more duplicates. The rest is covered by the last catch-all request matcher.

I don't think that would work. The curly brackets mark the regex. In your example, you use the regex outside of the brackets would become part of the path matching, not the regex, same goes for the brackets.

`/api/{version:(v\\d+/)?}public/**` might work, but honestly, it gets confusing. Having two of these makes it pretty clear that one is for the default option and one is for the versioned options.